java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,343 @@
+                        dateMilliVector.setValueCount(rowCount + 1);
+                        if (date != null) {
+                            dateMilliVector.setSafe(rowCount, rs.getDate(i).getTime());
+                        } else {
+                            dateMilliVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.TIME:
+                        Time time = rs.getTime(i);
+                        TimeMilliVector timeMilliVector = (TimeMilliVector)root.getVector(columnName);
+                        timeMilliVector.setValueCount(rowCount + 1);
+                        if (time != null) {
+                            timeMilliVector.setSafe(rowCount, (int) rs.getTime(i).getTime());
+                        } else {
+                            timeMilliVector.setNull(rowCount);
+                        }
+
+                        break;
+                    case Types.TIMESTAMP:
+                        // timezone is null
+                        Timestamp timestamp = rs.getTimestamp(i);
+                        TimeStampVector timeStampVector = (TimeStampVector)root.getVector(columnName);
+                        timeStampVector.setValueCount(rowCount + 1);
+                        if (timestamp != null) {
+                            timeStampVector.setSafe(rowCount, timestamp.getTime());
+                        } else {
+                            timeStampVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.BINARY:
+                    case Types.VARBINARY:
+                    case Types.LONGVARBINARY:
+                        VarBinaryVector varBinaryVector = (VarBinaryVector)root.getVector(columnName);;
+                        varBinaryVector.setValueCount(rowCount + 1);
+                        byte[] bytes = rs.getBytes(i);
+                        if (bytes != null) {
+                            varBinaryVector.setIndexDefined(rowCount);
+                            varBinaryVector.setValueLengthSafe(rowCount, bytes.length);
+                            varBinaryVector.setSafe(rowCount, bytes);
+                        } else {
+                            varBinaryVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.ARRAY:
+                        // not handled
+//                    fields.add(new Field("list", FieldType.nullable(new ArrowType.List()), null));
+                        break;
+                    case Types.CLOB:
+                        VarCharVector varcharVector1 = (VarCharVector)root.getVector(columnName);
+                        varcharVector1.setValueCount(rowCount + 1);
+                        Clob clob = rs.getClob(i);
+                        if (clob != null) {
+                            int length = (int) clob.length();
```

Comment:
if the value is bigger than `Integer.MAX_VALUE`, might cause to return an invalid value, with an exception if the value is negative, or truncated if the value is positive...